### PR TITLE
Set joined=message.clock on syncing pubchat creation

### PR DESCRIPTION
**Pull request by rasom**
needed for https://github.com/status-im/status-react/issues/12913

In case if current timestamp is set to `chat.joined` but there is a newer `SyncChatRemoved` message, the latter might be ignored as `chat.joined` bigger than `message.clock`. 

status: ready

### CODE REVIEW

**Review by Parveshdhull — APPROVED** [review]

**Review by bitgamma — APPROVED** [review]

File: protocol/messenger_handler.go [hunk 1]
```diff
@@ -421,8 +421,9 @@ func (m *Messenger) HandleSyncInstallationPublicChat(state *ReceivedMessageState
 	chat := existingChat
 	if !ok {
 		chat = CreatePublicChat(chatID, state.Timesource)
+		chat.Joined = int64(message.Clock)
 	} else {
-		existingChat.Joined = int64(state.Timesource.GetCurrentTime())
+		existingChat.Joined = int64(message.Clock)
```
Comment (Parveshdhull) [inline protocol/messenger_handler.go:426]: chat.Joined ?

**Review by Parveshdhull — CHANGES_REQUESTED** [review]

File: protocol/messenger_handler.go (hunk 1 above)
Comment (rasom) [reply protocol/messenger_handler.go:426]: what do you mean?

Comment (Parveshdhull) [reply protocol/messenger_handler.go:426]: Oh sorry, I should have included more details.

We are modifying `existingChat.Joined` but we are not using it 🤔, should we also change this into `chat.Joined`? then we can just also remove if else check.

**Review by Parveshdhull — CHANGES_REQUESTED** [review]

File: protocol/messenger_handler.go (hunk 1 above)
Comment (Parveshdhull) [reply protocol/messenger_handler.go:426]: Oh sorry, I should have included more details.

We are modifying `existingChat.Joined` but we are not using it 🤔, should we change this into `chat.Joined` ? then, we can also remove if else check

Comment (rasom) [reply protocol/messenger_handler.go:426]: both are pointers to the same value, so I doubt there is any difference

**Review by Parveshdhull — APPROVED** [review]

**Pull request merged**